system/jlib/jptree.cpp

```diff
@@ -6635,13 +6635,18 @@ class CJSONReader : public CJSONReaderBase<X>, implements IPTreeReader
                 iEvent->newAttribute(name, value.str());
             return;
         }
-        else if ('#'==*name)
+        else if ('#'==*name) // binary
         {
-            dbgassertex(retValue);
+            dbgassertex(retValue && isValueBinary);
+            *isValueBinary = false;
             bool iptValue = 0 == strncmp(name+1, "value", 5); // this is a special IPT JSON prop name, representing a 'complex' value
             if (iptValue)
             {
-                retValue->swapWith(value);
+                *isValueBinary = 0 == strncmp(name+6, "bin", 3);
```

Comment:
would it be more explicit (i.e. not match "#valuebinabc") to just do?:

`*isValueBinary = streq(name+6, "bin");`

Reply:
agree, will change.